build_daemon/lib/client.dart

```diff
@@ -17,8 +17,13 @@ import 'data/build_target_request.dart';
 import 'data/serializers.dart';
 import 'data/server_log.dart';
 
-int _existingPort(String workingDirectory) {
+Future<int> _existingPort(String workingDirectory) async {
   var portFile = File(portFilePath(workingDirectory));
+  var retryCount = 0;
+  while (!portFile.existsSync() && retryCount < maxRetries) {
```

Comment:
Thinking about this more we should probably model it as a timeout instead just to reduce the potential for accidentally increasing the amount of time we will wait.

Essentially, make `maxRetries` instead be a `Duration`, and use a stopwatch to bail out if the elapsed time > maxDuration.

Reply:
I don't know if it's necessary here since the work is synchronous and we only have async for the delay - but if we need it here's a retry function that we could either stick here or in `package:aynsc`. https://gist.github.com/natebosch/b038dd928e7bc9292d592b841f277471

Reply:
TIL: `Future.timeout` is a thing :D